stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/FailoverInitEndpoint.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http:www.apache.orglicensesLICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.net.http.websocket.client;
+
+import org.ballerinalang.jvm.scheduling.Strand;
+import org.ballerinalang.jvm.values.ArrayValue;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+import org.ballerinalang.net.http.HttpUtil;
+import org.ballerinalang.net.http.websocket.WebSocketConstants;
+import org.ballerinalang.net.http.websocket.WebSocketException;
+import org.ballerinalang.net.http.websocket.WebSocketUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+/**
+ * Initialize the WebSocket Client.
+ *
+ * @since 1.1.0
+ */
+
+@BallerinaFunction(
+        orgName = WebSocketConstants.BALLERINA_ORG,
+        packageName = WebSocketConstants.PACKAGE_HTTP,
+        functionName = "init",
+        receiver = @Receiver(
+                type = TypeKind.OBJECT,
+                structType = WebSocketConstants.FAILOVER_WEBSOCKET_CLIENT,
+                structPackage = WebSocketConstants.FULL_PACKAGE_HTTP
+        )
+)
+public class FailoverInitEndpoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(FailoverInitEndpoint.class);
+
+    public static void init(Strand strand, ObjectValue webSocketClient) throws URISyntaxException {
+        @SuppressWarnings(WebSocketConstants.UNCHECKED)
+        MapValue<String, Object> clientEndpointConfig = (MapValue<String, Object>) webSocketClient.getMapValue(
+                WebSocketConstants.CLIENT_ENDPOINT_CONFIG);
+        ArrayValue targets = clientEndpointConfig.getArrayValue(WebSocketConstants.TARGET_URLS);
+        ArrayList<String> newTargetUrls = new ArrayList<>();
+        int index = 0;
+        // check whether url has valid format or not
+        // if It isn't in the valid format, remove that from the url set
+        for (int i = 0; i < targets.size(); i++) {
+            URI uri = new URI(targets.get(i).toString());
+            String scheme = uri.getScheme();
+            if (!"ws".equalsIgnoreCase(scheme) && !"wss".equalsIgnoreCase(scheme)) {
+                logger.error(targets.get(i).toString() + " drop from the targets url" +
```

Comment:
Shall we have the loggers in all places to use the buit-in formatting? This is a sonar code smell
https://rules.sonarsource.com/java/RSPEC-2629

Reply:
Also refer https://stackoverflow.com/a/48163845/3599535